DS13/code/modules/overmap/warpcore.dm

```diff
@@ -0,0 +1,676 @@
+//This contains all the parts for the warp engine
+/*
+
+Activation sequence:
+Complete the deuterium + anti deuterium injectors (they start semi finished)
+Open deut + anti deut transmission manifolds to feed the injectors
+Set transmission ratio such that it�s 0.5 : 0.5 for a stable reaction
+Add dilithium to dilithium matrix
+Realign dilithium matrix to your desired spec
+Insert dilithium matrix into core
+Activate matter + antimatter stream (turn on injectors)
+Monitor the reaction for instability
+
+*/
+/obj/machinery/proc/default_radial_toggle(var/mob/user)
+	if(!allowed(user))
+		visible_message("<span class='notice'>You need an engineering level ID to access this machine.</span>")
+		return
+	var/list/options = list("on", "off")
+	for(var/option in options)
+		options[option] = image(icon = 'DS13/icons/actions/engine_actions.dmi', icon_state = "[option]")
+	var/dowhat = show_radial_menu(user,src,options)
+	if(!dowhat)
+		return
+	return dowhat
+
+/obj/machinery/power/matter_injector
+	name = "matter stream injector"
+	desc = "A huge device used to inject matter into a controlled antimatter reaction system for the purpose of power generation."
+	icon = 'DS13/icons/obj/power/warpcore/injectors.dmi'
+	icon_state = "injector-1"
+	anchored = TRUE
+	density = TRUE
+	var/active = FALSE
+	var/obj/machinery/power/warp_core/core //We need one of these.
+	var/injects_matter = TRUE //set to false for antimatter injector
+	var/construction_state = "cables"
+	var/can_activate = FALSE //Are we built and ready?
```

Comment:
Relying on `can_activate` as a flag is bug prone if someone somewhere forgets to change its state. Perhaps it would be better to have a `can_activate` procedure that gives you a response based on checking the states of the obj?